src/rust/api/dns.rs

```diff
@@ -0,0 +1,354 @@
+        let service_offset = flag_offset + flag_length + 1;
+        let service = decode_hex(&data[service_offset..service_length + service_offset])?.join("");
+
+        let regexp_length = usize::from_str_radix(data[service_offset + service_length], 16)?;
+        let regexp_offset = service_offset + service_length + 1;
+        let regexp = decode_hex(&data[regexp_offset..regexp_length + regexp_offset])?.join("");
+
+        let replacement = parse_replacement(&data[regexp_offset + regexp_length..])?;
+
+        Ok(NaptrRecord {
+            flags,
+            service,
+            regexp,
+            replacement,
+            order,
+            preference,
+        })
+    }
+}
+
+// Generated code.
+impl DnsUtil {
+    extern "C" fn parse_caa_record_callback(args: *mut jsg::v8::ffi::FunctionCallbackInfo) {
+        let mut lock = unsafe { jsg::v8::Lock::from_args(args) };
+        let len = unsafe { jsg::v8::ffi::get_length(args) };
+        assert!(len > 0);
+        let arg0 = unsafe { jsg::v8::ffi::get_arg(args, 0) };
+        let arg0 = unsafe { jsg::v8::ffi::unwrap_string(lock.get_isolate(), arg0) };
+        let args = unsafe { jsg::v8::FunctionCallbackInfo::from_ffi(args) };
+        let this = args.get_this(&mut lock);
+        let self_ = jsg::unwrap_resource::<DnsUtil>(&mut lock, this);
+        match self_.parse_caa_record(&arg0) {
+            Ok(record) => args.set_return_value(record.wrap(&mut lock)),
+            Err(err) => {
+                dbg!(err);
+            }
+        };
+    }
+
+    extern "C" fn parse_naptr_record_callback(args: *mut jsg::v8::ffi::FunctionCallbackInfo) {
+        let mut lock = unsafe { jsg::v8::Lock::from_args(args) };
+        let len = unsafe { jsg::v8::ffi::get_length(args) };
+        assert!(len > 0);
+        let arg0 = unsafe { jsg::v8::ffi::get_arg(args, 0) };
+        let arg0 = unsafe { jsg::v8::ffi::unwrap_string(lock.get_isolate(), arg0) };
+        let args = unsafe { jsg::v8::FunctionCallbackInfo::from_ffi(args) };
+        let this = args.get_this(&mut lock);
+        let self_ = jsg::unwrap_resource::<DnsUtil>(&mut lock, this);
+        match self_.parse_naptr_record(&arg0) {
+            Ok(record) => args.set_return_value(record.wrap(&mut lock)),
+            Err(err) => {
+                dbg!(err);
+            }
+        };
+    }
+}
+
+// Generated code.
+impl jsg::Resource for DnsUtil {
+    fn members() -> Vec<jsg::Member<Self>>
+    where
+        Self: Sized,
+    {
+        vec![
+            jsg::Member::Method {
+                name: "parseCaaRecord",
+                callback: Self::parse_caa_record_callback,
+            },
+            jsg::Member::Method {
+                name: "parseNaptrRecord",
+                callback: Self::parse_naptr_record_callback,
+            },
+        ]
+    }
+
+    fn class_name() -> &'static str {
+        "DnsUtil"
+    }
+
+    fn js_instance<'a>(&self, lock: &mut v8::Lock) -> Option<v8::LocalValue<'a>> {
```

Comment:
The `v8::Lock` here is misleading. We should keep the `v8::` prefix only for things that actually come from v8.